### DISCUSSION TITLE
Total number of solutions is not same with the same input

### DISCUSSION THREAD

**Discussion by sarthakgaur** [google/or-tools#2877] [CP-SAT questions] [answered]
Language: Python
Solver: CP-SAT

We have a problem similar to the [employee scheduling](https://developers.google.com/optimization/scheduling/employee_scheduling) problem. With the same input sometimes I get 42 total solutions and sometimes I get 38 total solutions. The input is exactly the same in both these cases.

Can someone explain why this is happening? Any help will be appreciated! Thanks!

**Comment by lperron** [comment]
Do you have an objective ?

**Reply by sarthakgaur** [reply]
Yes, we do have an objective. Our problem is more similar to [scheduling with shift requests](https://developers.google.com/optimization/scheduling/employee_scheduling#requests) example.

**Comment by lperron** [comment, marked as answer]
So the number of solutions found until reaching the optimal solutions is
random.
Remember that the optimal solver will produce only one solution.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le sam. 30 oct. 2021 à 15:36, sarthakgaur ***@***.***> a
écrit :
